cuda/Simulation.cu

```diff
@@ -26,16 +27,27 @@ unsigned long long run_event_based_simulation_baseline(Inputs in, SimulationData
         int nthreads = 256;
         int nblocks = ceil( (double) in.lookups / (double) nthreads);
 
-
-        xs_lookup_kernel_baseline<<<nblocks, nthreads>>>( in, GSD );
-        gpuErrchk( cudaPeekAtLastError() );
+	int nwarmups = in.num_iterations / 10;
+	start = 0.0;
+	for (int i = 0; i < in.num_iterations + nwarmups; i++) {
+		if (i == nwarmups) {
+			gpuErrchk( cudaDeviceSynchronize() );
+			start = get_time();
+		}
+		xs_lookup_kernel_baseline<<<nblocks, nthreads>>>( in, GSD );
+	}
+	gpuErrchk( cudaPeekAtLastError() );
+	gpuErrchk( cudaDeviceSynchronize() );
+	profile->kernel_time = get_time() - start;
 
         ////////////////////////////////////////////////////////////////////////////////
         // Reduce Verification Results
         ////////////////////////////////////////////////////////////////////////////////
 
         if( mype == 0)	printf("Reducing verification results...\n");
+	start = get_time();
         gpuErrchk(cudaMemcpy(SD.verification, GSD.verification, in.lookups * sizeof(unsigned long), cudaMemcpyDeviceToHost) );
+	profile->d2h_time = get_time() - start;
```

Comment:
This seems only be getting the final kernel time? I thought we wanted to keep track of all kernel time iterations?